builtin/profiler/instrumentation.lua

```diff
@@ -45,66 +47,128 @@ local register_functions = {
 	register_on_player_hpchange = 0,
 }
 
+local registered_instruments = {}
 ---
--- Create an unique instrument name.
--- Generate a missing label with a running index number.
---
-local counts = {}
-local function generate_name(def)
-	local class, label, func_name = def.class, def.label, def.func_name
-	if label then
-		if class or func_name then
-			return format("%s '%s' %s", class or "", label, func_name or ""):trim()
+-- Represents and identifies the instrumenting component, its metadata and logged quantities.
+-- An instrumented component may either be an instrumented function or a collection of instruments.
+-- Effectively creating a flat hierachy of instruments (usually with the top representing the entirety of instruments for a mod)
+-- References are passed around the profiler to avoid table lookups to find the instruments for logging,
+-- moving search and sort from instrumentation to sampling time for better performance.
```

Comment:
These comments are quite long. Could you please limit each line to 80 (maximal 90) characters counting from the first non-whitespace?